src/styles/components/_calendar.scss

```diff
@@ -0,0 +1,50 @@
+@import "../base/base";
+@import "../abstracts/variables";
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $width;
+  gap: $gap;
+  padding: $padding;
+
+  @for $i from 1 through 31 {
+    &__day-#{$i} {
+      background-color: $cell-color;
+      width: $cell-dimensions;
+      height: $cell-dimensions;
+      box-sizing: border-box;
+      border: 1px solid black;
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      cursor: pointer;
+    }
+    &__day-#{$i}:hover {
+      background-color: pink;
+      transform: $cell-animation;
+      animation-duration: 0.5s;
```

Comment:
You want to use transition, not animation - you don't have any keyframes.

Instead add transition to `calendar__day` (not hover). Remember to specify which property you want to apply transition to. Also use a shorthand and add transition duration there